jupyter-extension/jupyterlab_pachyderm/mount_server_client.py

```diff
@@ -187,18 +187,60 @@ async def auth_login(self):
         # may bubble exception up to handler if oidc_state not in response
         oidc = resp_json['oidc_state']
 
+        # we explicitly send the login_token request and do not await here.
+        # the reason for this is that we want the user to be redirected to
+        # the login page without awaiting the result of the login before
+        # doing so.
         asyncio.create_task(self.auth_login_token(oidc))
         return response.body
 
     async def auth_login_token(self, oidc):
         response = await self.client.fetch(
             f"{self.address}/auth/_login_token", method="PUT", body=f'{oidc}'
         )
-        os.makedirs(os.path.dirname(PACH_CONFIG_DIR), exist_ok=True)
-        with open(PACH_CONFIG_DIR, 'w') as f:
-            f.write(response.body.decode())
+        response.rethrow()
+        pach_config_dir = Path.home().joinpath('.pachyderm', 'config.json')
+        if pach_config_dir.is_file():
+            # if config already exists, need to add new context into it and
+            # switch active context over
+            try:
+                self.write_token_to_config(pach_config_dir,
+                                           response.body.decode())
+            except Exception as e:
+                get_logger().warn("Failed writing session token: ", e.args)
+                raise e
+        else:
+            # otherwise, write the entire config to file
+            os.makedirs(os.path.dirname(pach_config_dir), exist_ok=True)
+            with open(pach_config_dir, 'w') as f:
+                f.write(response.body.decode())
+
         return response.body
 
+    # updates the pachyderm config with the session token of the mount server
+    # config. this will try to insert the token into the mount_server context
+    # if it already exists, or copies the mount server config if it doesn't.
+    # then, it switches the active context over to the mount_server context.
+    def write_token_to_config(self, pach_config_dir, mount_server_config_str):
```

Comment:
it appears based on its usage that `pach_config_dir` is a file? Should it be named differently? Nonetheless it would be helpful to have a docstring that documents these arguments. The above comment can be moved into the docstrnig as well